Examples/Io/Root/src/RootVertexPerformanceWriter.cpp

```diff
@@ -181,36 +201,189 @@ ActsExamples::ProcessCode ActsExamples::RootVertexPerformanceWriter::writeT(
   ACTS_INFO("Total number of detector-accepted truth primary vertices : "
             << m_nVtxDetAcceptance);
 
-  // Read track-associated truth particle input collection
-  const auto& associatedTruthParticles =
-      ctx.eventStore.get<SimParticleContainer>(
-          m_cfg.inputAssociatedTruthParticles);
-  // Get number of track-associated true primary vertices
-  m_nVtxReconstructable =
-      getNumberOfReconstructableVertices(associatedTruthParticles);
-
-  ACTS_INFO("Total number of reco track-associated truth particles in event : "
-            << associatedTruthParticles.size());
-  ACTS_INFO("Total number of reco track-associated truth primary vertices : "
-            << m_nVtxReconstructable);
-
-  /*****************  Start x,y,z resolution plots here *****************/
-  // Matching tracks at vertex to fitted tracks that are in turn matched
-  // to truth particles. Match reco and true vtx if >50% of tracks match
-
   const auto& inputFittedTracks =
       ctx.eventStore.get<std::vector<Acts::BoundTrackParameters>>(
           m_cfg.inputFittedTracks);
 
   ACTS_INFO(
       "Total number of reconstructed tracks : " << inputFittedTracks.size());
 
-  if (associatedTruthParticles.size() != inputFittedTracks.size()) {
+  if (!m_cfg.inputAssociatedTruthParticles.empty()) {
+    // Read track-associated truth particle input collection
+    const auto& associatedTruthParticles =
+        ctx.eventStore.get<SimParticleContainer>(
+            m_cfg.inputAssociatedTruthParticles);
+    // Get number of track-associated true primary vertices
+    m_nVtxReconstructable =
+        getNumberOfReconstructableVertices(associatedTruthParticles);
+
     ACTS_INFO(
-        "Number of fitted tracks and associated truth particles do not match. "
-        "Not able to match fitted tracks at reconstructed vertex to truth "
-        "vertex.");
-  } else {
+        "Total number of reco track-associated truth particles in event : "
+        << associatedTruthParticles.size());
+    ACTS_INFO("Total number of reco track-associated truth primary vertices : "
+              << m_nVtxReconstructable);
+
+    /*****************  Start x,y,z resolution plots here *****************/
+    // Matching tracks at vertex to fitted tracks that are in turn matched
+    // to truth particles. Match reco and true vtx if >50% of tracks match
+
+    if (associatedTruthParticles.size() != inputFittedTracks.size()) {
+      ACTS_INFO(
```

Comment:
would an error be more appropriate here?

Reply:
Now that it supports the other case, maybe yeah.

Reply:
Ok I checked again, and this still triggers in the case of <1 track efficiency e.g. in truth tracking. I added conditional ERROR or INFO messages.